docs/API.md

```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
 
 But can be short written as:
 ```js
-stamp = stamp.static({
+stamp = stamp.statics({
   foo: 'foo'
 });
 ```
 
-See more useful tips in the [advanced examples](advanced_examples.md#validate-before-a-function-call).
-
-## Utility methods
-
-### stampit.methods()
-
-Shortcut for `stampit().methods()`
-
-### stampit.refs()
-
-Shortcut for `stampit().refs()` 
-
-### stampit.init()
-
-Shortcut for `stampit().init()`
 
-### stampit.props()
+### stamp.deepStatics() and stamp.deepStaticProperties()
 
-Shortcut for `stampit().props()`
+Same as `stamp.statics()` and `stamp.staticProperties()` but deeply merges the
+provided objects.
 
-### stampit.compose()
 
-Shortcut for `stampit().compose()`
+### stamp.conf() and stamp.configuration()
 
-### stampit.mixin(), .extend(), .mixIn(), .assign()
+Take n objects and add all its properties to the stamp's metadata.
 
-Aliases to `Object.assign()`. Deprecated.
+* `@return {Object} stamp` The new stamp based on the original `this` stamp.
 
+#### Examples
 
-### stampit.isStamp(obj)
+Use metadata in intializers:
+```js
+const stamp = stampit()
+.conf({addFactorSetter: false})
+.init((opts, {stamp}) => {
+  let factor = opts.factor || 1;
+  instance.getFactor = () => factor;
+  
+  if (stamp.compose.configuration.addFactorSetter) {
+    instance.setFactor = f => factor = f;    
+  }
+});
 
-Take an object and return `true` if it's a stamp, `false` otherwise.
+console.log(stamp().setFactor); // undefined
+const stamp2 = stamp.conf({addFactorSetter: false}); 
+console.log(stamp2(5).getFactor()); // 5
+```
 
+Use metadata in static functions:
+```js
+const stamp = stampit()
+.statics({
+  allowFactorSetter(allow) {
+    return this.conf({addFactorSetter: !!allow})
+  }
+})
+.init((opts, {stamp}) => {
+  let factor = opts.factor || 1;
+  instance.getFactor = () => factor;
+  
+  if (stamp.compose.configuration.addFactorSetter) {
+    instance.setFactor = f => factor = f;    
+  }
+});
 
-### stampit.convertConstructor()
+console.log(stamp().setFactor); // undefined
+const stamp2 = stamp.allowFactorSetter(true); 
+console.log(stamp2().setFactor(5).getFactor()); // 5
+```
 
-Take an old-fashioned JS constructor and return a stamp  that
-you can freely compose with other stamps. It is possible to
-use constructors that take arguments. Simply pass the arguments
-into the returned stamp after the properties object:
-`const myInstance = myStamp(props, arg1, arg2);`
 
-Note that if you use this feature, it is **not safe** to compose
-the resulting stamp with other stamps willy-nilly, because if two
-different stamps depend on the argument passing feature, the arguments
-will probably clash with each other, producing very unexpected results.
+### stamp.deepConf() and stamp.deepConfiguration()
 
- * @param  {Function} Constructor
- * @return {Function} A composable stampit factory (aka stamp).
+Same as `stamp.conf()` and `stamp.configuration()` but deeply merges the
+provided objects.
```

Comment:
Might be worth mentioning here that result is accessible in `stamp.compose.deepConfiguration`. That's another difference and might cause confusion people wondering why they don't see `configuration` correctly.